src/dbnode/storage/shard_test.go

```diff
@@ -493,7 +495,61 @@ func TestShardTick(t *testing.T) {
 	_, ok := shard.flushState.statesByTime[xtime.ToUnixNano(earliestFlush)]
 	require.True(t, ok)
 }
+
+type testWrite struct {
+	id         string
+	value      float64
+	unit       xtime.Unit
+	annotation []byte
+}
+
 func TestShardWriteAsync(t *testing.T) {
+	testShardWriteAsync(t, []testWrite{
+		{
+			id:    "foo",
+			value: 1.0,
+			unit:  xtime.Second,
+		},
+		{
+			id:    "bar",
+			value: 2.0,
+			unit:  xtime.Second,
+		},
+		{
+			id:    "baz",
+			value: 3.0,
+			unit:  xtime.Second,
+		},
+	})
+}
+
+func TestShardWriteAsyncWithAnnotations(t *testing.T) {
+	testShardWriteAsync(t, []testWrite{
```

Comment:
one more test case - ensure we don't copy the annotation bytes if the series already exists in the shard